grails-app/controllers/au/org/ala/biocollect/ReferenceAssessmentController.groovy

```diff
@@ -0,0 +1,178 @@
+        def refDoc = referenceActivity.documents[0]
+
+        def baseUrl = ''
+        if (!refDoc["url"].startsWith('http')) {
+            baseUrl = grailsApplication.config.grails.serverURL
+        }
+
+        def assessPhoto = [
+                licence: refDoc["licence"],
+                notes: refDoc["notes"],
+                filesize: refDoc["filesize"],
+                staged: true,
+                url: baseUrl + refDoc["url"],
+                filename: refDoc["filename"],
+                attribution: referenceActivity.outputs[0].data["imageAttribution"],
+                name: refDoc["name"],
+                documentId: '',
+                contentType: refDoc["contentType"],
+                dateTaken: refDoc["dateTaken"],
+                formattedSize: refDoc["formattedSize"],
+                thumbnailUrl: baseUrl + refDoc["thumbnailUrl"],
+                status: "active"
+        ]
+
+        def assessActivity = [
+                outputs: [
+                        [
+                                outputId: "",
+                                outputNotCompleted: false,
+                                data: [
+                                        recordedBy: userService.getCurrentUserDisplayName(),
+                                        upperConditionBound: "0",
+                                        lowerConditionBound: "0",
+                                        overallConditionBestEstimate: "0",
+                                        mvgGroup: referenceActivity.outputs[0].data.vegetationStructureGroup,
+                                        huchinsonGroup: referenceActivity.outputs[0].data.huchinsonGroup,
+                                        sitePhoto: [assessPhoto],
+                                        deIdentify: deIdentify ? "Yes" : "No"
+                                ],
+                                name: assessProjectActivity["pActivityFormName"]
+                        ]
+                ],
+                projectActivityId: assessProjectActivity["projectActivityId"],
+                userId: userService.getCurrentUserId(),
+                projectStage: "",
+                embargoed: false,
+                type: assessProjectActivity["pActivityFormName"],
+                projectId: assessProjectActivity["projectId"],
+                mainTheme: ""
+        ]
+
+        // Create the new assessment activity record
+        activityService.update("", assessActivity)
+
+        // Update the numTimesReferenced field on the reference record
+        referenceActivity.outputs[0].data.numTimesReferenced =
+                referenceActivity.outputs[0].data.numTimesReferenced as Integer + 1
+        activityService.update(referenceActivity.activityId, referenceActivity)
+
+        // Return the assessment activity
+        assessActivity
+    }
+
+    def requestRecords() {
+        def config = grailsApplication.config.refAssess
```

Comment:
Use `getProperty` to get config items. `grailsApplication.config.getProperty("refAssess", Map)`. A few instances of this are in the code

Reply:
Fixed